### PR TITLE
Added example of using libsigc++ with Qt

### PR DESCRIPTION
**Pull request by rm5248**

### CODE REVIEW

**Comment by murraycu** [comment]
Thanks, but what is this example showing us?

Is it showing us how to call a libsigc++ slot from a Qt signal handler (slot)? If so, is that particularly difficult to figure out, or difficult to do when also using Qt? Isn't it just like calling a function?

Or is it just an example of how you need to disable the Qt "keywords", instead using the Q_* macros?

**Comment by rm5248** [comment]
> Is it showing us how to call a libsigc++ slot from a Qt signal handler (slot)? If so, is that particularly difficult to figure out, or difficult to do when also using Qt? Isn't it just like calling a function?

There's not a big difference between the Qt signals/slots and libsigc++, but they can be used in a similar manner.

> Or is it just an example of how you need to disable the Qt "keywords", instead using the Q_* macros?

This is the needed part.  I also wanted to show that you can use both the Qt signals/slots at the same time as libsigc++.

Because of that, it is a relatively trivial example, as otherwise it could get very complicated.

**Comment by murraycu** [comment]
OK. Thanks. Then I'd like it to be clearer about that, please. I think it should be as minimal as possible - it should have Qt signals and slots and libsigc++ signals and slots, just to show how they can co-exist within the same class (because of the use of the Q_* keywords) but they should not interact.

And I think there should be a README.md saying what it is showing.

**Comment by rm5248** [comment]
OK, I've added a README.md and simplified the example slightly.  The signals don't interact with one another much; the slot that is used for sigc++ is also a Qt slot, but I feel that it is important to mention.  The one thing that the example does that is not a good practice is emitting signals inside the constructor; would it make sense to have that in a separate method for clarity?

File: examples/qt/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# libsigc++ with Qt
+
+It is possible to use libsigc++ with Qt.  However, because of the signals/slots
+mechanism of Qt, there is some setup that must be done in order for this to
+happen correctly.
+
+Steps to use libsigc++ with Qt:
+1. In your .pro file, add `CONFIG += no_keywords`.  This configures Qt to not
```
Comment (murraycu) [inline examples/qt/README.md]: This assumes that the person is using qmake, I think. But Qt doesn't require the use of qmake.

File: examples/qt/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+define the keywords `emit`, `signals`, and `slot`.
```
Comment (murraycu) [inline examples/qt/README.md]: These are macros, not keywords.

File: examples/qt/README.md (hunk 1 above)
Comment (murraycu) [reply examples/qt/README.md]: Also, I think it's worth saying that this (using the Q_* macros) is generally good advice when writing Qt code.

Comment (rm5248) [reply examples/qt/README.md]: I'm generally in agreement with that, I'll go update this in a bit.  While I don't disagree that using the Q_* macros is a good idea, that's generally not what happens in most Qt code that I have seen(even Qt's own examples use `emit`/`signals`/`slots` and not the Q_* macros).

File: examples/qt/README.md (hunk 2 above)
Comment (rm5248) [reply examples/qt/README.md]: They're macros in the C++ land, but that is paraphrasing what the Qt documentation says in the link you put above: 

> It tells Qt not to define the moc keywords `signals`, `slots`, and `emit` [...]

**Comment by rm5248** [comment]
Updated based off of feedback.

**Comment by murraycu** [comment]
Many thanks, and sorry for being so slow to respond.

I have squashed these into one commit and pushed them to master.

This directory is not DISTed into the tarball (by make distcheck), and we should fix that, but it's not fatal.

**Comment by kjellahl** [comment]
>This directory is not DISTed into the tarball (by make distcheck)

It's not distributed by `make distcheck`. It *is* distributed by `ninja dist`.
Meson/ninja distributes all files from the git repo, unless instructions in a
meson.build file removes some files from the meson-dist directory.

The example with Qt is not built by a meson build or an autotools build. If it's
ever added to the build systems, building it should be controlled by a configuration
option, and disabled by default, like the benchmark with Boost.

**Pull request closed without merging**